### ISSUE TITLE
`Style/MutableConstant` with `EnforcedStyle: strict` false positive on `Data.define`

### ISSUE THREAD

**Issue by RYOTA-KOBA** [rubocop/rubocop#15284]
## Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

## Issue type

False positive

## Cop name

`Style/MutableConstant`

## Steps to reproduce the problem

1. Create a file `example.rb` with the following content:

```ruby
# frozen_string_literal: true

MyData = Data.define(:name, :age)
```

2. Run:

```bash
rubocop example.rb --only Style/MutableConstant
```

## Expected behavior

No offense should be reported. `Data` (introduced in Ruby 3.2) is a DSL for defining immutable value classes whose instances are always frozen:

```ruby
Data.define(:name, :age).new(name: "Alice", age: 30).frozen? # => true
```

While the class object returned by `Data.define` is itself not frozen (just like `Struct.new`), it is used to declare an immutable value type rather than as a mutable container that needs explicit freezing. This is analogous to how `Struct.new` is already treated as immutable by this cop.

## Actual behavior

With `EnforcedStyle: strict`, RuboCop reports:

```
example.rb:3:12: C: Style/MutableConstant: Freeze mutable objects assigned to constants.
MyData = Data.define(:name, :age)
         ^^^^^^^^^^^^^^^^^^^^^^^^
```

## RuboCop configuration

```yaml
Style/MutableConstant:
  EnforcedStyle: strict
```

## RuboCop version

```
1.87.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.3.10) +Rubydex [arm64-darwin25]
```
